tests/component/sections/collection/collection-items-panel/CollectionItemsPanel.spec.tsx

```diff
@@ -0,0 +1,443 @@
+          typesQuery: undefined
+        }}
+        addDataSlot={null}
+      />
+    )
+
+    cy.findByTestId('items-list').should('exist').children().should('have.length', 10)
+    cy.findByTestId('collection-items-list-infinite-scroll-skeleton').should('exist')
+  })
+
+  it('renders 10 first items and then loads more items when scrolling to the bottom', () => {
+    cy.customMount(
+      <CollectionItemsPanel
+        collectionId={ROOT_COLLECTION_ALIAS}
+        collectionRepository={collectionRepository}
+        collectionQueryParams={{
+          pageQuery: 1,
+          searchQuery: undefined,
+          typesQuery: undefined
+        }}
+        addDataSlot={null}
+      />
+    )
+
+    cy.findByTestId('items-list').should('exist').children().should('have.length', 10)
+    cy.findByTestId('collection-items-list-infinite-scroll-skeleton').should('exist')
+
+    cy.findByTestId('items-list-scrollable-container').scrollTo('bottom')
+
+    cy.findByTestId('items-list').should('exist').children().should('have.length', 20)
+    cy.findByTestId('collection-items-list-infinite-scroll-skeleton').should('exist')
+  })
+
+  it('renders 4 items with no more to load, correct results in header, and no bottom skeleton loader', () => {
+    const first4Elements = items.slice(0, 4)
+    const first4ElementsWithCount: CollectionItemSubset = {
+      items: first4Elements,
+      totalItemCount: 4
+    }
+    collectionRepository.getItems = cy.stub().resolves(first4ElementsWithCount)
+
+    cy.customMount(
+      <CollectionItemsPanel
+        collectionId={ROOT_COLLECTION_ALIAS}
+        collectionRepository={collectionRepository}
+        collectionQueryParams={{
+          pageQuery: 1,
+          searchQuery: undefined,
+          typesQuery: undefined
+        }}
+        addDataSlot={null}
+      />
+    )
+
+    cy.findByText('4 results').should('exist')
+    cy.findByTestId('items-list').should('exist').children().should('have.length', 4)
+    cy.findByTestId('collection-items-list-infinite-scroll-skeleton').should('not.exist')
+  })
+
+  it('sets Collections and Datasets as default selected when no types query param is passed', () => {
+    cy.customMount(
+      <CollectionItemsPanel
+        collectionId={ROOT_COLLECTION_ALIAS}
+        collectionRepository={collectionRepository}
+        collectionQueryParams={{
+          pageQuery: 1,
+          searchQuery: undefined,
+          typesQuery: undefined
+        }}
+        addDataSlot={null}
+      />
+    )
+
+    cy.findByRole('checkbox', { name: /Collections/ }).should('be.checked')
+    cy.findByRole('checkbox', { name: /Datasets/ }).should('be.checked')
+    cy.findByRole('checkbox', { name: /Files/ }).should('not.be.checked')
+  })
+
+  /*
+    The things that happened inside the handleSearchSubmit and the handleItemsTypeChange function is not currently possible to test, will be tested in e2e tests
+    Adding this so it goes through the test coverage
+  */
```

Comment:
Just curious, why is it not possible to test? Is there a problem mocking the API calls?

Reply:
These mentioned functions do a lot of things inside the function body and end up updating the URL query params. Actually in Component tests you can't change the URL query params in Cypress, so it's only possible to assert in e2e tests that the function update the URL correctly.